Emotion 10 intrinsics typescript types do not have event handler props

<!--
Thanks for your interest in the project. I appreciate bugs filed and PRs submitted!
Please make sure that you are familiar with and follow the Code of Conduct for
this project (found in the CODE_OF_CONDUCT.md file).

Please fill out this template with all the relevant information so we can
understand what's going on and fix the issue.

I'll probably ask you to submit the fix (after giving some direction). If you've
never done that before, that's great! Check this free short video tutorial to
learn how: http://kcd.im/pull-request
-->

- `emotion` version: 10.0.4
- `react` version: 16.7.0-alpha.2

**Relevant code:**

```javascript
const Form = styled.form({})

const Foo = () => <Form onSubmit={e => { console.log(e) }} />
```


**What you did:**

Upgraded to Emotion 10

**What happened:**

Intrinsics created by `styled` do not appear to have event handler props like `onSubmit`/`onChange` so typescript complains about the implicit any of `e`.

```
error TS7006: Parameter 'event' implicitly has an 'any' type.
```

**Reproduction:**

Codesandbox doesn't import emotion types, so it's not helpful to repro. If there's a way to make that work, or if there's a starter project w/ Typescript + emotion 10 I'd gladly whip up a repro.

--- a/packages/core/types/index.d.ts
+++ b/packages/core/types/index.d.ts
@@ -19,6 +19,7 @@ import {
 export {
   ArrayInterpolation,
   ComponentSelector,
+  CSSObject,
   FunctionInterpolation,
   ObjectInterpolation
 } from '@emotion/css'

--- a/packages/css/types/index.d.ts
+++ b/packages/css/types/index.d.ts
@@ -5,6 +5,7 @@ import { Interpolation, SerializedStyles } from '@emotion/serialize'
 export {
   ArrayInterpolation,
   ComponentSelector,
+  CSSObject,
   FunctionInterpolation,
   ObjectInterpolation
 } from '@emotion/serialize'

--- a/packages/serialize/types/index.d.ts
+++ b/packages/serialize/types/index.d.ts
@@ -22,6 +22,31 @@ export interface CSSOthersObject<MP> {
   [propertiesName: string]: Interpolation<MP>
 }
 
+export type CSSPseudosForCSSObject = { [K in CSS.Pseudos]?: CSSObject }
+
+export interface ArrayCSSInterpolation extends Array<CSSInterpolation> {}
+
+export type CSSInterpolation =
+  | null
+  | undefined
+  | boolean
+  | number
+  | string
+  | ComponentSelector
+  | Keyframes
+  | SerializedStyles
+  | CSSObject
+  | ArrayCSSInterpolation
+
+export interface CSSOthersObjectForCSSObject {
+  [propertiesName: string]: CSSInterpolation
+}
+
+export interface CSSObject
+  extends CSSPropertiesWithMultiValues,
+    CSSPseudosForCSSObject,
+    CSSOthersObjectForCSSObject {}
+
 export interface ComponentSelector {
   __emotion_styles: any
 }

--- a/packages/serialize/types/tests.ts
+++ b/packages/serialize/types/tests.ts
@@ -1,4 +1,5 @@
 import {
+  CSSObject,
   ObjectInterpolation,
   Keyframes,
   serializeStyles
@@ -52,3 +53,13 @@ serializeStyles()
 serializeStyles({})
 // $ExpectError
 serializeStyles({}, {})
+
+let cssObject: CSSObject = {
+  fontWeight: 400,
+  ':hover': {
+    fontWeight: 700
+  }
+}
+
+// $ExpectError
+cssObject = { fontWeight: 'wrong' }

--- a/packages/styled-base/types/index.d.ts
+++ b/packages/styled-base/types/index.d.ts
@@ -19,6 +19,7 @@ import { Omit, Overwrapped, PropsOf } from './helper'
 
 export {
   ArrayInterpolation,
+  CSSObject,
   FunctionInterpolation,
   ObjectInterpolation
 } from '@emotion/serialize'

--- a/packages/styled/types/index.d.ts
+++ b/packages/styled/types/index.d.ts
@@ -12,6 +12,7 @@ export {
   CreateStyledComponentBase,
   CreateStyledComponentExtrinsic,
   CreateStyledComponentIntrinsic,
+  CSSObject,
   FunctionInterpolation,
   Interpolation,
   ObjectInterpolation,

